src/hostapi/alsa/pa_linux_alsa.c

```diff
@@ -4616,12 +4616,14 @@ static PaError GetAlsaStreamPointer( PaStream* s, PaAlsaStream** stream )
 
 PaError PaAlsa_GetStreamInputCard( PaStream* s, int* card )
 {
-    PaAlsaStream *stream;
+    PaAlsaStream *stream = NULL;
     PaError result = paNoError;
     snd_pcm_info_t* pcmInfo;
 
     PA_ENSURE( GetAlsaStreamPointer( s, &stream ) );
 
+    if (stream == NULL) return paBadStreamPtr;
```

Comment:
The paBadStreamPtr error is mostly for when the caller passes in a NULL pointer.
This error is not the caller's fault so maybe use paUnanticipatedHostError instead.

Reply:
Feel free to change the error number.

Reply:
The NULL check is redundant.
We should set stream=NULL right before the call because that is the PA pattern.

Reply:
The NULL check would catch a problem if GetAlsaStreamPointer would report OK, but not set the stream. Not sure whether this could happen.

Reply:
@MonkeyBreadSoftware: if GetAlsaStreamPointer returns paNoError but does not return a valid stream pointer it is badly broken. That's not the kind of thing that we check at runtime.